streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -330,6 +330,20 @@ abstract class DStream[T: ClassTag] (
   }
 
   /**
+   * compute the last valid time till input time
+   * (validTime = zeroTime + N * slideDuration)
+   */
+  private[streaming] def lastValidTime(time: Time): Time = {
+    if (time < zeroTime) {
+      throw new IllegalArgumentException(
+          s"Input(${time}) must bigger then zeroTime($zeroTime)")
+    }
+
+    val multi: Int = ((time - zeroTime) / slideDuration).floor.toInt
```

Comment:
They do the same thing, but I would still like to keep `floor` method explicitly instead of delegating to default float->int convention, in order to make it clearer and avoiding potential risk.